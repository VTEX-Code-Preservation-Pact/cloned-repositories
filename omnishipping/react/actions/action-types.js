export const CHANGE_ACTIVE_TAB = 'CHANGE_ACTIVE_TAB'
export const CHANGE_ACTIVE_SLA_OPTION = 'CHANGE_ACTIVE_SLA_OPTION'
export const SELECT_DELIVERY_WINDOW = 'SELECT_DELIVERY_WINDOW'

export const CHANGE_ACTIVE_PICKUP_OPTIONS = 'CHANGE_ACTIVE_PICKUP_OPTIONS'
export const CHANGE_ACTIVE_PICKUP_SELLER = 'CHANGE_ACTIVE_PICKUP_SELLER'
export const CHANGE_ACTIVE_DELIVERY_CHANNEL = 'CHANGE_ACTIVE_DELIVERY_CHANNEL'
export const CLOSE_PICKUP_MODAL = 'CLOSE_PICKUP_MODAL'
export const OPEN_PICKUP_MODAL = 'OPEN_PICKUP_MODAL'

export const CHANGE_ACTIVE_PICKUP_DETAILS = 'CHANGE_ACTIVE_PICKUP_DETAILS'
export const CLOSE_PICKUP_DETAILS = 'CLOSE_PICKUP_DETAILS'

export const UPDATE_ADDRESS_FORM = 'UPDATE_ADDRESS_FORM'
export const UPDATE_ADDRESS_RULES = 'UPDATE_ADDRESS_RULES'
export const CHANGE_SELECTED_ADDRESS_RULES = 'CHANGE_SELECTED_ADDRESS_RULES'
export const VALIDATE_ADDRESS_FORM = 'VALIDATE_ADDRESS_FORM'
export const VALIDATE_ADDRESS_FORM_FIELDS = 'VALIDATE_ADDRESS_FORM_FIELDS'
export const VALIDATE_POSTAL_CODE = 'VALIDATE_POSTAL_CODE'
export const ADD_ADDRESS_FORM_ADDRESS = 'ADD_ADDRESS_FORM_ADDRESS'
export const ADD_DELIVERY_ADDRESS = 'ADD_DELIVERY_ADDRESS'
export const UPDATE_ORDERFORM = 'UPDATE_ORDERFORM'

export const SELECT_ADDRESS_ITEM = 'SELECT_ADDRESS_ITEM'

export const EDIT_ADDRESS = 'EDIT_ADDRESS'
export const EDIT_OMNISHIPPING = 'EDIT_OMNISHIPPING'
export const NEW_ADDRESS = 'NEW_ADDRESS'
export const SHOW_ADDRESS_PRODUCTS = 'SHOW_ADDRESS_PRODUCTS'
export const SHOW_ADDRESS = 'SHOW_ADDRESS'
export const SHOW_SPLIT = 'SHOW_SPLIT'
export const SHOW_ADDRESS_LIST = 'SHOW_ADDRESS_LIST'
export const SHOW_PAYMENT_BUTTON = 'SHOW_PAYMENT_BUTTON'
export const SHOW_OMNISHIPPING_COMPLETED = 'SHOW_OMNISHIPPING_COMPLETED'
export const SHOW_OMNISHIPPING = 'SHOW_OMNISHIPPING'
export const ENABLE_OMNISHIPPING = 'ENABLE_OMNISHIPPING'
export const DISABLE_OMNISHIPPING = 'DISABLE_OMNISHIPPING'

export const UPDATE_SHIPPING_FAILURE = 'UPDATE_SHIPPING_FAILURE'
export const UPDATE_SHIPPING_FROM_SHIPPING_RATE_PREVIEW =
  'UPDATE_SHIPPING_FROM_SHIPPING_RATE_PREVIEW'
export const UPDATE_SHIPPING_REQUEST = 'UPDATE_SHIPPING_REQUEST'

export const SIMULATE_SHIPPING_REQUEST = 'SIMULATE_SHIPPING_REQUEST'
export const SIMULATE_SHIPPING_FAILURE = 'SIMULATE_SHIPPING_FAILURE'
export const SIMULATE_SHIPPING_SUCCESS = 'SIMULATE_SHIPPING_SUCCESS'

export const UPDATE_SIMULATION_OPTIONS = 'UPDATE_SIMULATION_OPTIONS'

export const REMOVE_ITEMS_REQUEST = 'REMOVE_ITEMS_REQUEST'

export const FOCUS_ON_INPUT = 'FOCUS_ON_INPUT'

export const HAS_PICKUP_OPTIONS_EVENT = 'HAS_PICKUP_OPTIONS_EVENT'
export const SELECTED_PICKUP_TAB_EVENT = 'SELECTED_PICKUP_TAB_EVENT'
export const SHIPPING_STATS_EVENT = 'SHIPPING_STATS_EVENT'
export const SEARCHED_PICKUP_ADDRESS_EVENT = 'SEARCHED_PICKUP_ADDRESS_EVENT'
export const SEARCHED_PICKUP_ADDRESS_GEOLOCATION_EVENT =
  'SEARCHED_PICKUP_ADDRESS_GEOLOCATION_EVENT'

export const VTEX_ID_STARTED = 'VTEX_ID_STARTED'
export const VTEX_ID_CLOSED = 'VTEX_ID_CLOSED'

export const CHANGE_ACTIVE_DELIVERY_PACKAGE = 'CHANGE_ACTIVE_DELIVERY_PACKAGE'

export const UPDATE_SELECTED_PACKAGE_SLIDER = 'UPDATE_SELECTED_PACKAGE_SLIDER'

export const HAS_GEOLOCATION = 'HAS_GEOLOCATION'

export const SELECT_SCHEDULED_DELIVERY = 'SELECT_SCHEDULED_DELIVERY'
export const TOGGLE_SCHEDULED_DELIVERY = 'TOGGLE_SCHEDULED_DELIVERY'

export const OMNISHIPPING_JS_ERROR = 'OMNISHIPPING_JS_ERROR'
export const OMNISHIPPING_API_ERROR = 'OMNISHIPPING_API_ERROR'

export const SET_ADITIONAL_SHIPPING_DATA = 'SET_ADITIONAL_SHIPPING_DATA'

export const OMNISHIPPING_PAGEVIEW_EVENT = 'OMNISHIPPING_PAGEVIEW_EVENT'

export const COMPLETE_OMNISHIPPING = 'COMPLETE_OMNISHIPPING'
