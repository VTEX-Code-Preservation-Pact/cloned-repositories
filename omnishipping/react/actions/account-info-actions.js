import * as types from './action-types'
import {
  getAccountName,
  getLocale,
  getUserEmail,
} from '../selectors/order-form-selectors'
import { AUTHENTICATED_USER } from '../constants/checkout-events'
const $ = window.$

export const vtexIdStarted = () => ({
  type: types.VTEX_ID_STARTED,
})

export const closedVtexId = () => ({
  type: types.VTEX_ID_CLOSED,
})

export const startVtexID = callback => (dispatch, getState) => {
  const state = getState()

  dispatch(vtexIdStarted())

  window.vtexid &&
    window.vtexid.start({
      returnUrl: window.location.href,
      accountName: getAccountName(state) || window.vtex.accountName,
      locale: getLocale(state),
      email: getUserEmail(state),
    })

  const authCallback = () => callback && callback(true)
  if ($) {
    $(window).one(AUTHENTICATED_USER, authCallback)
    $(window).one('closed.vtexid', () => {
      dispatch(closedVtexId())
      $(window).off(AUTHENTICATED_USER, authCallback)
    })
  }
}
