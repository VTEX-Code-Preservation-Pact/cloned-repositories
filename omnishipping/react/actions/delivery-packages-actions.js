import * as types from './action-types'

export const updateSelectedPackageSlider = packageIndex => ({
  type: types.UPDATE_SELECTED_PACKAGE_SLIDER,
  packageIndex,
})
