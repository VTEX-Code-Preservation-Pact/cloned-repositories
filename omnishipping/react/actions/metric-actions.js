import map from 'lodash/map'
import * as types from './action-types'
import { getAllSelectedPackages } from '../selectors/delivery-selectors'
import {
  CHEAPEST,
  FASTEST,
  VTEXBETA,
  VTEXSTABLE,
  VTEXLOCAL,
  INFO,
  IMPORTANT,
  ERROR,
  CRITICAL,
} from '../constants'
import { isPickup } from '../utils/DeliveryChannelsUtils'
import SplunkEvents from 'splunk-events'
import EvidenceClient from '@vtex/evidence-client-js'
import axios from 'axios'
const splunkEvents = new SplunkEvents()
const evidenceClient = new EvidenceClient()
let omnishippingVersion = null

setTimeout(() => {
  omnishippingVersion = getIOAppVersion({
    appName: 'omnishipping',
    vendor: 'vtex',
  })
}, 500)

splunkEvents.config({
  endpoint: 'https://splunk-heavyforwarder-public.vtex.com:8088',
  token: '50fe94b0-30b6-442a-9cb1-a476c97ba917',
  request: axios,
})

evidenceClient.config({
  request: axios,
})

function getIOAppVersion({ appName, vendor }) {
  const IOAppName = Object.keys(window.__RENDER_7_COMPONENTS__).find(
    key => key.indexOf(appName) > 0 && key.indexOf('.x') === -1
  )

  return (
    IOAppName &&
    IOAppName.replace(`${vendor}.${appName}@`, '').replace('/index', '')
  )
}

function sendEvidence(evidence = {}) {
  return evidenceClient.sendEvidence('checkout-ui', JSON.stringify(evidence))
}

export const omnishippingJsError = (error, componentStack, id) => ({
  type: types.OMNISHIPPING_JS_ERROR,
  error,
  componentStack,
  id,
})

export const omnishippingAPIError = (
  errorMessage,
  operationId,
  apiRoute,
  reduxState,
  apiMessage,
  errorStatus,
  requestPayload,
  url,
  id
) => ({
  type: types.OMNISHIPPING_API_ERROR,
  errorMessage,
  operationId,
  apiRoute,
  reduxState,
  apiMessage,
  errorStatus,
  requestPayload,
  url,
  id,
})

export const searchPickupAddressByGeolocation = () => ({
  type: types.SEARCHED_PICKUP_ADDRESS_GEOLOCATION_EVENT,
})

export const hasPickup = () => ({
  type: types.HAS_PICKUP_OPTIONS_EVENT,
})

export const selectedPickupTab = () => ({
  type: types.SELECTED_PICKUP_TAB_EVENT,
})

export const searchedPickupAddress = () => ({
  type: types.SEARCHED_PICKUP_ADDRESS_EVENT,
})

export const omnishippingPageViewEvent = omnishippingVersion => ({
  type: types.OMNISHIPPING_PAGEVIEW_EVENT,
  omnishippingVersion,
})

export const shippingStats = () => ({
  type: types.SHIPPING_STATS_EVENT,
})

export const searchPickupAddressByGeolocationEvent = data => dispatch => {
  dispatch(searchPickupAddressByGeolocation())

  log({
    name: 'searchPickupAddressByGeolocation',
    data: {
      env: isEnv(VTEXBETA) ? VTEXBETA : VTEXSTABLE,
      searchedAddressByGeolocation: data.searchedAddressByGeolocation || false,
      confirmedGeolocationPopUp: data.confirmedGeolocation || false,
      dismissedGeolocation: data.dismissedGeolocation || false,
      deniedGeolocationPopUp: data.deniedGeolocation || false,
      positionUnavailable: data.positionUnavailable || false,
      browserError: data.browserError || false,
    },
  })
}

export const allowe = () => dispatch => {
  dispatch(hasPickup())

  log({
    name: 'searchPickupAddressByGeolocation',
    data: {
      env: isEnv(VTEXBETA) ? VTEXBETA : VTEXSTABLE,
    },
  })
}

export const hasPickupEvent = () => dispatch => {
  dispatch(hasPickup())

  log({
    name: 'hasPickup',
    data: {
      env: isEnv(VTEXBETA) ? VTEXBETA : VTEXSTABLE,
    },
  })
}

export const selectedPickupEvent = () => dispatch => {
  dispatch(selectedPickupTab())

  log({
    name: 'selectedPickupTab',
    data: {
      env: isEnv(VTEXBETA) ? VTEXBETA : VTEXSTABLE,
    },
  })
}

export const searchedPickupAddressEvent = () => dispatch => {
  dispatch(searchedPickupAddress())

  log({
    name: 'searchedPickupAddress',
    data: {
      env: isEnv(VTEXBETA) ? VTEXBETA : VTEXSTABLE,
    },
  })
}

export const omnishippingJsErrorEvent = ({
  error,
  componentStack,
  errorId,
  reduxState,
}) => (dispatch, getState) => {
  dispatch(omnishippingJsError(error, componentStack, errorId))
  const evidenceHash = sendEvidence(reduxState)
  const accountName = getState().accountInfo.accountName

  logSplunk({
    level: CRITICAL,
    type: ERROR,
    workflowType: 'shipping',
    workflowInstance: 'js-error',
    event: {
      errorMessage: error.message,
      errorStack: error.stack,
      componentStack,
      errorId,
      evidenceHash,
      accountName,
      version: omnishippingVersion,
    },
  })

  log({
    name: 'omnishippingJsError',
    data: {
      errorMessage: error.message,
      errorStack: error.stack,
      componentStack,
      errorId,
      evidenceHash,
    },
  })
}

export const omnishippingPageView = () => dispatch => {
  dispatch(omnishippingPageViewEvent(omnishippingVersion))

  log({
    name: 'omnishippingVersion',
    data: {
      omnishippingVersion,
    },
  })
}

export const omnishippingAPIErrorEvent = ({
  apiMessage,
  apiRoute,
  errorMessage,
  errorStatus,
  operationId,
  reduxState,
  requestPayload,
  url,
}) => (dispatch, getState) => {
  dispatch(
    omnishippingAPIError(
      apiMessage,
      apiRoute,
      errorMessage,
      errorStatus,
      operationId,
      reduxState,
      requestPayload,
      url
    )
  )

  const evidenceHash = sendEvidence(reduxState)
  const accountName = getState().accountInfo.accountName

  logSplunk({
    level: CRITICAL,
    type: ERROR,
    workflowType: 'shipping',
    workflowInstance: 'omnishipping-api-error',
    event: {
      apiMessage,
      apiRoute,
      errorMessage,
      errorStatus,
      operationId,
      evidenceHash,
      requestPayload,
      url,
      accountName,
      version: omnishippingVersion,
    },
  })

  log({
    name: 'omnishippingAPIError',
    data: {
      apiMessage,
      apiRoute,
      errorStatus,
      errorMessage,
      operationId,
      evidenceHash,
      requestPayload,
      url,
    },
  })
}

export const shippingStatsEvent = () => (dispatch, getState) => {
  const state = getState()
  const packages = getAllSelectedPackages(state)
  const activeDeliveryOption = state.delivery.activeDeliveryOption
  const hasCheapestOption = !!state.delivery[CHEAPEST]
  const hasFastestOption = !!state.delivery[FASTEST]

  dispatch(shippingStats())

  map(packages, liPackage => {
    logSplunk({
      level: IMPORTANT,
      type: INFO,
      workflowType: 'shipping',
      workflowInstance: 'shipping-stats',
      event: {
        hasPickupAndChosePickup: isPickup(liPackage),
        selectedDeliveryChannel: liPackage.deliveryChannel,
        packages: packages.length,
        pickupSelectionSource: isPickup(liPackage) ? 'map' : null,
        chosenOption: activeDeliveryOption,
        isSingleOption: !(hasCheapestOption && hasFastestOption),
        hasChosenScheduledDelivery: liPackage.deliveryWindow !== null,
        version: omnishippingVersion,
      },
    })

    log({
      name: 'shippingStats',
      data: {
        env: isEnv(VTEXBETA) ? VTEXBETA : VTEXSTABLE,
        hasPickupAndChosePickup: isPickup(liPackage),
        selectedDeliveryChannel: liPackage.deliveryChannel,
        packages: packages.length,
        pickupSelectionSource: isPickup(liPackage) ? 'map' : null,
        chosenOption: activeDeliveryOption,
        isSingleOption: !(hasCheapestOption && hasFastestOption),
        hasChosenScheduledDelivery: liPackage.deliveryWindow !== null,
      },
    })
  })
}

function logSplunk(config) {
  const { level, type, workflowType, workflowInstance, event, account } = config

  /* istanbul ignore else */
  if (!isEnv(VTEXLOCAL)) {
    splunkEvents.logEvent(
      level,
      type,
      workflowType,
      workflowInstance,
      event,
      account
    )
  } else {
    console.log('Log', config)
  }
}

function log(data) {
  /* istanbul ignore else */
  if (!isEnv(VTEXLOCAL)) {
    window.checkoutLogger && window.checkoutLogger(data)
  } else {
    console.log('Log', data)
  }
}

function isEnv(env) {
  return window.location.href.indexOf(env) !== -1
}
