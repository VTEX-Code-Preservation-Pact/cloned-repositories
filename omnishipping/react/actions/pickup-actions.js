import * as types from './action-types'
import { DENIED, VTEXLOCAL } from '../constants'

export const changeActivePickupOptions = pickupOptions => ({
  type: types.CHANGE_ACTIVE_PICKUP_OPTIONS,
  pickupOptions,
})

export const changeActivePickupDetails = ({
  pickupPoint,
  activePickupPoint,
  askForGeolocation,
}) => ({
  type: types.CHANGE_ACTIVE_PICKUP_DETAILS,
  pickupPoint,
  activePickupPoint,
  askForGeolocation,
})

export const closePickupModal = () => ({
  type: types.CLOSE_PICKUP_MODAL,
})

export const openPickupModal = ({ askForGeolocation }) => ({
  type: types.OPEN_PICKUP_MODAL,
  askForGeolocation,
})

export const closePickupDetails = () => ({
  type: types.CLOSE_PICKUP_DETAILS,
})

export const changeActivePickupSeller = sellerId => ({
  type: types.CHANGE_ACTIVE_PICKUP_SELLER,
  sellerId,
})

export const hasGeolocation = hasGeolocation => ({
  type: types.HAS_GEOLOCATION,
  hasGeolocation,
})

export const checkIfHasDeniedGeolocation = () => dispatch => {
  if (!navigator.permissions) {
    dispatch(hasGeolocation(!!navigator.geolocation))
  } else {
    navigator.permissions
      .query({ name: 'geolocation' })
      .then(permission =>
        dispatch(
          hasGeolocation(
            permission.state !== DENIED ||
              window.location.host.includes(VTEXLOCAL)
          )
        )
      )
  }
}
