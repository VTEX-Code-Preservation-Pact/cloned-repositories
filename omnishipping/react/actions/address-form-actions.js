import * as types from './action-types'
import { newAddress } from '../utils/AddressFormUtils'
import {
  getDefaultShipsTo,
  getStoreCountryCode,
} from '../selectors/order-form-selectors'
import { getSearchAddress } from '../selectors/address-form-selectors'
import { newAddressEvent } from './component-activity-actions'
import { SEARCH } from '../constants'
import { helpers } from 'vtex.address-form'
const { removeValidation } = helpers

export const updateAddressForm = ({
  address,
  searchAddress,
  channel,
  rules,
}) => ({
  type: types.UPDATE_ADDRESS_FORM,
  address,
  searchAddress,
  channel,
  rules,
})

export const validateAddressForm = rules => ({
  type: types.VALIDATE_ADDRESS_FORM,
  rules,
})

export const validateAddressFormFields = rules => ({
  type: types.VALIDATE_ADDRESS_FORM_FIELDS,
  rules,
})

export const validatePostalCode = rules => ({
  type: types.VALIDATE_POSTAL_CODE,
  rules,
})

export const addAddressFormAddress = ({ address, searchAddress, channel }) => ({
  type: types.ADD_ADDRESS_FORM_ADDRESS,
  address,
  searchAddress,
  channel,
})

export const focusOnInput = input => ({
  type: types.FOCUS_ON_INPUT,
  input,
})

export const addDeliveryAddress = () => (dispatch, getState) => {
  dispatch(newAddressEvent(true))
  const state = getState()
  const shipsTo = getDefaultShipsTo(state)
  const storeCountryCode = getStoreCountryCode(state)
  const channel = state.componentActivity.activeDeliveryChannel
  const searchAddress = removeValidation(getSearchAddress(state))
  const country =
    shipsTo.indexOf(storeCountryCode) > -1 ? storeCountryCode : shipsTo[0]
  dispatch(
    addAddressFormAddress({
      address: newAddress({ country }),
      searchAddress,
      channel,
    })
  )
}

export const addNewAddresses = () => (dispatch, getState) => {
  const state = getState()
  const shipsTo = getDefaultShipsTo(state)
  const storeCountryCode = getStoreCountryCode(state)
  const channel = state.componentActivity.activeDeliveryChannel
  const country =
    shipsTo.indexOf(storeCountryCode) > -1 ? storeCountryCode : shipsTo[0]
  dispatch(
    addAddressFormAddress({
      address: newAddress({ country }),
      searchAddress: newAddress({
        country,
        addressType: SEARCH,
      }),
      channel,
    })
  )
}
