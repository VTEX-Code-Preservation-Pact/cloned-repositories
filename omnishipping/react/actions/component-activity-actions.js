import * as types from './action-types'
import { setItem } from '../utils/LocalStorageUtils'
import { OMNISHIPPING } from '../constants'
import { SHIPPING_INFO_UPDATED } from '../constants/checkout-events'
import {
  changeActiveTab,
  updateShippingData,
} from '../actions/order-form-actions'
import { isPickup } from '../utils/DeliveryChannelsUtils'
import {
  selectedPickupEvent,
  shippingStatsEvent,
} from '../actions/metric-actions'
import { validateAddressFormFields } from './address-form-actions'
import { shouldCompleteOmnishipping } from '../selectors/omnishipping-selectors'
import { getShippingDataWithSelectedDelivery } from '../selectors/delivery-selectors'
import { isValidAddress } from '../selectors/address-form-selectors'

const $ = window.$

export const selectDeliveryChannel = channel => (dispatch, getState) => {
  const state = getState()

  const activeDeliveryChannel = state.componentActivity.activeDeliveryChannel

  if (channel === activeDeliveryChannel) return

  dispatch(changeActiveDeliveryChannel(channel))
  dispatch(
    setAditionalShippingData({
      activeTab: channel,
    })
  )
  dispatch(
    changeActiveTab({
      channel: channel,
      loading: false,
      shouldUpdate: true,
    })
  )

  if (isPickup(channel)) {
    dispatch(selectedPickupEvent())
  }
}

export const enableOmnishipping = () => ({
  type: types.ENABLE_OMNISHIPPING,
})

export const disableOmnishipping = () => ({
  type: types.DISABLE_OMNISHIPPING,
})

export const showAddressList = show => ({
  type: types.SHOW_ADDRESS_LIST,
  show,
})

export const showAddress = show => ({
  type: types.SHOW_ADDRESS,
  show,
})

export const showSplit = show => ({
  type: types.SHOW_SPLIT,
  show,
})

export const showOmniShippingCompleted = show => ({
  type: types.SHOW_OMNISHIPPING_COMPLETED,
  show,
})

export const showOmniShipping = show => ({
  type: types.SHOW_OMNISHIPPING,
  show,
})

export const editOmnishipping = edit => ({
  type: types.EDIT_OMNISHIPPING,
  edit,
})

export const editAddress = show => ({
  type: types.EDIT_ADDRESS,
  show,
})

export const newAddressEvent = show => ({
  type: types.NEW_ADDRESS,
  show,
})

export const changeActiveDeliveryChannel = deliveryChannel => ({
  type: types.CHANGE_ACTIVE_DELIVERY_CHANNEL,
  deliveryChannel,
})

export const setAditionalShippingDataAction = () => ({
  type: types.SET_ADITIONAL_SHIPPING_DATA,
})

export const completeOmnishippingAction = () => ({
  type: types.COMPLETE_OMNISHIPPING,
})

export const completeOmnishipping = rules => (dispatch, getState) => {
  const state = getState()
  const shouldComplete = shouldCompleteOmnishipping(state)
  const shippingData = getShippingDataWithSelectedDelivery(state)

  dispatch(completeOmnishippingAction())
  if (!isValidAddress(state)) {
    dispatch(validateAddressFormFields(rules))
  }

  if (shouldComplete) {
    dispatch(showOmniShippingCompleted(true))
    dispatch(editOmnishipping(false))
    dispatch(updateShippingData(shippingData))
    dispatch(shippingStatsEvent())
  }
}

export const setAditionalShippingData = ({
  activeTab,
  selectedLeanShippingOption,
  isScheduledDeliveryActive,
}) => (dispatch, getState) => {
  const state = getState()

  const localActiveTab =
    activeTab || state.componentActivity.activeDeliveryChannel
  const localSelectedLeanShippingOption =
    selectedLeanShippingOption || state.delivery.activeDeliveryOption
  const localIsScheduledDeliveryActive =
    isScheduledDeliveryActive || state.delivery.isScheduledDeliveryActive

  const aditionalShippingData = {
    activeTab: localActiveTab,
    selectedLeanShippingOption: localSelectedLeanShippingOption,
    isScheduledDeliveryActive: localIsScheduledDeliveryActive,
    originComponent: OMNISHIPPING,
  }

  dispatch(setAditionalShippingDataAction(aditionalShippingData))

  setItem('aditionalShippingData', JSON.stringify(aditionalShippingData))

  $ && $(window).trigger(SHIPPING_INFO_UPDATED)
}
