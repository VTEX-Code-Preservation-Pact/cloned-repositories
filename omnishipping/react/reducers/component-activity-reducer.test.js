import reducer from './component-activity-reducer'
import * as types from '../actions/action-types'

describe('ComponentActivity reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle SHOW_OMNISHIPPING_COMPLETED', () => {
    expect(
      reducer([], {
        type: types.SHOW_OMNISHIPPING_COMPLETED,
        isAddressValid: true,
        show: true,
      })
    ).toMatchSnapshot()
  })

  it('should handle SHOW_OMNISHIPPING', () => {
    expect(
      reducer([], {
        type: types.SHOW_OMNISHIPPING,
        isAddressValid: true,
        show: true,
      })
    ).toMatchSnapshot()
  })
})
