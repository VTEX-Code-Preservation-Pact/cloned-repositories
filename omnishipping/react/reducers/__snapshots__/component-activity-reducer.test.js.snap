// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentActivity reducer should handle SHOW_OMNISHIPPING 1`] = `
Object {
  "isOmniShippingOpen": true,
}
`;

exports[`ComponentActivity reducer should handle SHOW_OMNISHIPPING_COMPLETED 1`] = `
Object {
  "editAddressActive": false,
  "isOmniShippingCompleted": true,
  "isSplitActive": true,
  "newAddressActive": false,
  "triedCompleteOmnishipping": false,
}
`;

exports[`ComponentActivity reducer should return the initial state 1`] = `
Object {
  "activeDeliveryChannel": "delivery",
  "editAddressActive": false,
  "hasPickup": false,
  "isAddressActive": false,
  "isAddressListActive": false,
  "isEditingOmnishipping": false,
  "isOmniShippingCompleted": false,
  "isOmniShippingOpen": false,
  "isSplitActive": false,
  "newAddressActive": false,
  "triedCompleteOmnishipping": false,
}
`;
